src/hotspot/share/gc/shenandoah/shenandoahOldGeneration.cpp

```diff
@@ -507,6 +507,58 @@ void ShenandoahOldGeneration::record_success_concurrent(bool abbreviated) {
   ShenandoahHeap::heap()->shenandoah_policy()->record_success_old();
 }
 
+void ShenandoahOldGeneration::trigger_collection_if_fragmented(ShenandoahGenerationalHeap* gen_heap, size_t first_old_region,
```

Comment:
As pointed out further above, would this more naturally belong here or in ShenandoahOldHeuristics? There are a bunch of triggers & checks there as well. Would appear to make sense to keep all of this in one place rather than scattering them between generation and heuristic for clear abstraction boundaries and maintainability of code.

Reply:
Agreed.  Thanks.  Moved.